### PR TITLE
Keep all key-value pairs when the document has duplicate key names.

### PR DESCRIPTION
**Pull request by rubcuevas**
Fixes: https://github.com/amzn/ion-python/issues/36

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: amazon/ion/core.py [hunk 1]
```diff
@@ -471,3 +474,87 @@ def timestamp(year, month=1, day=1,
...
+    as the built-in dict. If multiple elements are already mapped to the key, ``[]`  will return
+    the newest one.
+
+    To map multiple elements to a key, use the ``add_item`` operation.
+    To retrieve all the values map to a key, use ``get_all_values``.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(Multimap, self).__init__()
+        self.store = {}
```
Comment (tgregg) [inline amazon/ion/core.py]: Should this be named `self.__store`, as it's not intended to be accessed publicly?

File: amazon/ion/core.py [hunk 2]
```diff
@@ -471,3 +474,87 @@ def timestamp(year, month=1, day=1,
...
+        for k, v in self.iteritems():
+            output.append((k, v))
+        return output
+
+
+class MultimapValue(MutableSequence):
+
+    def __init__(self, *args):
+        if args is not None:
+            self.store = [x for x in args]
```
Comment (tgregg) [inline amazon/ion/core.py]: Should this be `self.__store` as it's not intended to be accessed publicly?

File: amazon/ion/core.py (hunk 1 above)
Comment (rubcuevas) [reply amazon/ion/core.py]: Addressed

File: amazon/ion/core.py (hunk 2 above)
Comment (rubcuevas) [reply amazon/ion/core.py]: Addressed

**Pull request merged**